--- a/Makefile
+++ b/Makefile
@@ -2,30 +2,14 @@
 KERNELRELEASE ?= $(shell uname -r)
 
 obj-m += cr14.o
-dtbo-y += cr14.dtbo
-
-targets += $(dtbo-y)
-
-# Gracefully supporting the new always-y without cutting off older target with kernel 4.x
-ifeq ($(firstword $(subst ., ,$(KERNELRELEASE))),4)
-	always := $(dtbo-y)
-else
-	always-y := $(dtbo-y)
-endif
-
-all:
-	make -C /lib/modules/$(KERNELRELEASE)/build M=$(PWD) modules
-
-# dtbo rule is no longer available
-ifeq ($(firstword $(subst ., ,$(KERNELRELEASE))),6)
 all: cr14.dtbo
+	$(MAKE) -C /lib/modules/$(KERNELRELEASE)/build M=$(shell pwd) modules
 
 cr14.dtbo: cr14-overlay.dts
-	dtc -I dts -O dtb -o $@ $<
-endif
+	dtc -@ -I dts -O dtb -o $@ $<
 
 clean:
-	make -C /lib/modules/$(KERNELRELEASE)/build M=$(PWD) clean
+	$(MAKE) -C /lib/modules/$(KERNELRELEASE)/build M=$(shell pwd) clean
 
 install: cr14.ko cr14.dtbo
 	install -o root -m 755 -d /lib/modules/$(KERNELRELEASE)/kernel/input/misc/
--- a/cr14.c
+++ b/cr14.c
@@ -827,7 +827,7 @@
 
 static void cr14_polling_timer_cb(struct timer_list *t)
 {
-	struct cr14_i2c_data *priv = from_timer(priv, t, polling_timer);
+	struct cr14_i2c_data *priv = timer_container_of(priv, t, polling_timer);
 	if (priv->opened) {
 		schedule_work(&priv->polling_work);
 	}
@@ -835,19 +835,19 @@
 
 static void restart_polling_timer(struct cr14_i2c_data *priv)
 {
-	del_timer_sync(&priv->polling_timer);
+	timer_delete_sync(&priv->polling_timer);
 	mod_timer(&priv->polling_timer,
 		  jiffies + HZ / POLLING_TIMEOUT_SECS_DIV);
 }
 
 static void stop_polling_timer(struct cr14_i2c_data *priv)
 {
-	del_timer_sync(&priv->polling_timer);
+	timer_delete_sync(&priv->polling_timer);
 }
 
 static void trigger_polling_work(struct cr14_i2c_data *priv)
 {
-	del_timer_sync(&priv->polling_timer);
+	timer_delete_sync(&priv->polling_timer);
 	if (priv->opened) {
 		schedule_work(&priv->polling_work);
 	}
@@ -1239,7 +1239,7 @@
 		unregister_chrdev_region(priv->chrdev, 2);
 	}
 
-	del_timer_sync(&priv->polling_timer);
+	timer_delete_sync(&priv->polling_timer);
 	cancel_work_sync(&priv->polling_work);
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(6, 0, 0)
