--- a/Makefile
+++ b/Makefile
@@ -2,22 +2,14 @@
 KERNELRELEASE ?= $(shell uname -r)
 
 obj-m += tagtagtag-ears.o
-dtbo-y += tagtagtag-ears.dtbo
+all: tagtagtag-ears.dtbo
+	$(MAKE) -C /lib/modules/$(KERNELRELEASE)/build M=$(shell pwd) modules
 
-targets += $(dtbo-y)
-
-# Gracefully supporting the new always-y without cutting off older target with kernel 4.x
-ifeq ($(firstword $(subst ., ,$(KERNELRELEASE))),4)
-	always := $(dtbo-y)
-else
-	always-y := $(dtbo-y)
-endif
-
-all:
-	make -C /lib/modules/$(KERNELRELEASE)/build M=$(PWD) modules
+tagtagtag-ears.dtbo: tagtagtag-ears-overlay.dts
+	dtc -@ -I dts -O dtb -o $@ $<
 
 clean:
-	make -C /lib/modules/$(KERNELRELEASE)/build M=$(PWD) clean
+	$(MAKE) -C /lib/modules/$(KERNELRELEASE)/build M=$(shell pwd) clean
 
 install: tagtagtag-ears.ko tagtagtag-ears.dtbo
 	install -o root -m 755 -d /lib/modules/$(KERNELRELEASE)/kernel/input/misc/
--- a/tagtagtag-ears.c
+++ b/tagtagtag-ears.c
@@ -30,6 +30,7 @@
 #include <linux/kernel.h>
 #include <linux/platform_device.h>
 #include <linux/of_device.h>
+#include <linux/of.h>
 #include <linux/gpio/consumer.h>
 #include <linux/fs.h>
 #include <linux/cdev.h>
@@ -146,7 +147,7 @@
 
 static int init_ear(struct device *dev, struct tagtagtagear_data *priv, struct class *ears_class, int major, int minor, const char* encoder_name, const char* motor_name);
 static int tagtagtagears_probe(struct platform_device *pdev);
-static int tagtagtagears_remove(struct platform_device *pdev);
+static void tagtagtagears_remove(struct platform_device *pdev);
 
 static int position_add(int position, int increment);
 
@@ -180,7 +181,7 @@
 // Always stop motors.
 //
 static void tagtagtagear_broken_timer_cb(struct timer_list *t) {
-    struct tagtagtagear_data *priv = from_timer(priv, t, broken_timer);
+    struct tagtagtagear_data *priv = timer_container_of(priv, t, broken_timer);
     stop_motors(priv);
     if (priv->state_e == testing) {
         dev_err(priv->device, "timeout, declaring ear as broken");
@@ -192,7 +193,7 @@
 }
 
 static void reset_broken_timer(struct tagtagtagear_data *priv) {
-    del_timer_sync(&priv->broken_timer);
+    timer_delete_sync(&priv->broken_timer);
     mod_timer(&priv->broken_timer, jiffies + BROKEN_TIMEOUT_SECS * HZ);
 }
 
@@ -259,7 +260,7 @@
         priv->state.running.direction = -1;
         start_motors_backward(priv);
     } else {
-        del_timer_sync(&priv->broken_timer);
+        timer_delete_sync(&priv->broken_timer);
         stop_motors(priv);  // We need to stop motors if we transitioned from detecting.
         if (priv->read_result_available == 1) {
             priv->read_result = position;
@@ -321,7 +322,7 @@
                 int ix;
 
                 // End of forward testing. Stop motors.
-                del_timer_sync(&priv->broken_timer);
+                timer_delete_sync(&priv->broken_timer);
                 stop_motors(priv);
                 // We should have 16 approximatively equivalent deltas and one at least twice larger.
                 first_delta = priv->state.testing.hole_deltas[0];
@@ -369,7 +370,7 @@
             int position;
             // We were running backward one position to test backward motor.
             // End of backward testing. Stop motors.
-            del_timer_sync(&priv->broken_timer);
+            timer_delete_sync(&priv->broken_timer);
             stop_motors(priv);
             if (priv->state.testing.forward_position == NUM_HOLES - EARS_OFFZERO) {
                 if (backward_delta < priv->detect_boundary_us) {
@@ -423,7 +424,7 @@
     priv->state.running.count--;
     if (priv->state.running.count == 0) {
         int is_high;
-        del_timer_sync(&priv->broken_timer);
+        timer_delete_sync(&priv->broken_timer);
         stop_motors(priv);
         is_high = gpiod_get_value(priv->encoder_gpio);
         if (is_high) {
@@ -885,7 +886,7 @@
     }
 
 	// Create device class
-	priv->ears_class = class_create(THIS_MODULE, DEVICE_NAME);
+	priv->ears_class = class_create(DEVICE_NAME);
 	if (IS_ERR(priv->ears_class)) {
 		err = PTR_ERR(priv->ears_class);
         dev_err(dev, "class_create failed: %d", err);
@@ -910,7 +911,7 @@
     return 0;
 }
 
-static int tagtagtagears_remove(struct platform_device *pdev) {
+static void tagtagtagears_remove(struct platform_device *pdev) {
     struct tagtagtagears_data *priv;
     int ix;
     priv = platform_get_drvdata(pdev);
@@ -919,7 +920,7 @@
         if (priv->ears_class) {
             for (ix = 1; ix >= 0; ix--) {
                 if (priv->ear[ix].cdev.ops) {
-                    del_timer_sync(&priv->ear[ix].broken_timer);
+                    timer_delete_sync(&priv->ear[ix].broken_timer);
                     device_destroy(priv->ears_class, MKDEV(MAJOR(priv->chrdev), MINOR(priv->chrdev) + ix));
                     cdev_del(&priv->ear[ix].cdev);
                 }
@@ -928,7 +929,6 @@
         }
         unregister_chrdev_region(priv->chrdev, 2);
     }
-    return 0;
 }
 
 #ifdef CONFIG_OF
